File: src/main/java/space/maxkonkin/nasapicbot/model/QueueMessage.java
package space.maxkonkin.nasapicbot.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@NoArgsConstructor
@Getter
@Setter
public class QueueMessage {
    @JsonProperty("messages")
    private List<Message> messages;

    @Getter
    @Setter
    public static class Message {
        @JsonProperty("event_metadata")
        private EventMetadata eventMetadata;
        @JsonProperty("details")
        private Details details;
    }

    @Getter
    @Setter
    public static class Details {
        @JsonProperty("queue_id")
        private String queueId;
        @JsonProperty("message")
        private MessageDetails message;
    }

    @Getter
    @Setter
    public static class MessageDetails {
        @JsonProperty("message_id")
        private String messageId;
        @JsonProperty("md5_of_body")
        private String md5OfBody;
        @JsonProperty("body")
        private String body;
        @JsonProperty("attributes")
        private Attributes attributes;
        @JsonProperty("message_attributes")
        private MessageAttributes messageAttributes;
        @JsonProperty("md5_of_message_attributes")
        private String md5OfMessageAttributes;
    }

    @Getter
    @Setter
    public static class Attributes {
        @JsonProperty("ApproximateFirstReceiveTimestamp")
        private String approximateFirstReceiveTimestamp;
        @JsonProperty("ApproximateReceiveCount")
        private String approximateReceiveCount;
        @JsonProperty("SenderId")
        private String senderId;
        @JsonProperty("SentTimestamp")
        private String sentTimestamp;
    }

    @Getter
    @Setter
    public static class EventMetadata {
        @JsonProperty("event_id")
        private String eventId;
        @JsonProperty("event_type")
        private String eventType;
        @JsonProperty("created_at")
        private String createdAt;
        @JsonProperty("tracing_context")
        private String tracingContext;
        @JsonProperty("cloud_id")
        private String cloudId;
        @JsonProperty("folder_id")
        private String folderId;
    }

    @Getter
    @Setter
    private static class MessageAttributes {
        @JsonProperty("messageAttributeKey")
        private MessageAttributeKey messageAttributeKey;
    }

    @Getter
    @Setter
    private static class MessageAttributeKey {
        @JsonProperty("dataType")
        private String dataType;
        @JsonProperty("stringValue")
        private String stringValue;
    }
}

File: src/main/kotlin/space/maxkonkin/nasapicbot/model/QueueMessage.kt
package space.maxkonkin.nasapicbot.model

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@JsonInclude(JsonInclude.Include.NON_EMPTY)
data class QueueMessage(
    @JsonProperty("messages")
    val messages: List<Message>
) {
    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    data class Message(
        @JsonProperty("event_metadata")
        val eventMetadata: EventMetadata?,
        @JsonProperty("details")
        val details: Details?
    )

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    data class Details(
        @JsonProperty("queue_id")
        val queueId: String?,
        @JsonProperty("message")
        val message: MessageDetails?
    )

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    data class MessageDetails(
        @JsonProperty("message_id")
        val messageId: String?,
        @JsonProperty("md5_of_body")
        val md5OfBody: String?,
        @JsonProperty("body")
        val body: String?,
        @JsonProperty("attributes")
        val attributes: Attributes?,
        @JsonProperty("message_attributes")
        val messageAttributes: MessageAttributes?,
        @JsonProperty("md5_of_message_attributes")
        val md5OfMessageAttributes: String?
    )

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    data class Attributes(
        @JsonProperty("ApproximateFirstReceiveTimestamp")
        val approximateFirstReceiveTimestamp: String?,
        @JsonProperty("ApproximateReceiveCount")
        val approximateReceiveCount: String?,
        @JsonProperty("SenderId")
        val senderId: String?,
        @JsonProperty("SentTimestamp")
        val sentTimestamp: String?
    )

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    data class EventMetadata(
        @JsonProperty("event_id")
        val eventId: String?,
        @JsonProperty("event_type")
        val eventType: String?,
        @JsonProperty("created_at")
        val createdAt: String?,
        @JsonProperty("tracing_context")
        val tracingContext: String?,
        @JsonProperty("cloud_id")
        val cloudId: String?,
        @JsonProperty("folder_id")
        val folderId: String?
    )

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    data class MessageAttributes(
        @JsonProperty("messageAttributeKey")
        val messageAttributeKey: MessageAttributeKey?
    )

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    data class MessageAttributeKey(
        @JsonProperty("dataType")
        val dataType: String?,
        @JsonProperty("stringvalue")
        val stringvalue: String?
    )
}
